doc/source/api/io.rst

```diff
@@ -0,0 +1,50 @@
+File-IO
```

Comment:
I would prefer to change the name of CUDS File into H5CUDS for the following reasons:
- The name is specific and help avoiding confusion with the FILE-IO based engine adapters
- The name allows alternative serialization types to be provided in the future (e.g SQLCUDS)

Then the section can be named accordingly